app/services/decidim/action_delegator/sms_gateway.rb

```diff
@@ -5,23 +5,44 @@ module ActionDelegator
     class SmsGateway
       attr_reader :mobile_phone_number, :code, :response
 
+      SMS_GATEWAY_PROVIDER_JOBS = {
+        som_connexio: SendSmsJob,
+        twilio: TwilioSendSmsJob
+      }.freeze
+
       def initialize(mobile_phone_number, code)
         @mobile_phone_number = mobile_phone_number
         @code = code
       end
 
       def deliver_code
-        SendSmsJob.perform_later(sender_name, mobile_phone_number, message)
+        return false unless sms_gateway_provider_valid?
+
+        sms_gateway_job.perform_later(sender_name, mobile_phone_number, message)
```

Comment:
As per my comment above and to avoid potential confusion, I'd rather rename `sender_name` to `sender` so it's abstract enough to "fit" both a name in case of SC or a phone number in the case of Twilio.